core/lib/Thelia/Module/BaseModule.php

```diff
@@ -435,7 +435,7 @@ public function getCurrentOrderTotalAmount($with_tax = true, $with_discount = tr
         /** @var Country $country */
         $country = $taxEngine->getDeliveryCountry();
 
-        $amount = $with_tax ? $cart->getTaxedAmount($country, $with_discount) : $cart->getTotalAmount($with_discount);
+        $amount = $with_tax ? $cart->getTaxedAmount($country, $with_discount) : $cart->getTotalAmount($with_discount, $country);
```

Comment:
As the state is also available through Thelia\TaxEngine\TaxEngine::getDeliveryState(), you can use it in the call to $cart->getTotalAmount()